crates/sui-indexer-alt-jsonrpc/src/api/coin.rs

```diff
@@ -169,7 +214,8 @@ impl CoinServer {
         &self,
         owner: SuiAddress,
         coin_type_tag: Option<TypeTag>,
-        page: Page<BalanceCursor>,
+        cursor: Option<Cursor<BalanceCursor>>,
```

Comment:
I don't feel great about an interface that claims to return a page, but where the limit is optional. Can we do the following instead:

- Factor out the code that generates the bulk of the select statement into its own function.
- Use it here, in a function that accepts a page, and returns a `PageResponse` (or `JsonRpcPage` as you have here -- it would be good to standardise this).
- Use it in another function that fetches it directly to get a vector of IDs (which you can use for `getAllBalances`).

Reply:
I gave it a try but it is in fact pretty tough, to me at least. The hard part is factoring out the select statement code into its own function -- the return type will have a very long name, something like `'BoxedSelectStatement<'
static, (Binary, BigInt, SmallInt), FromClause<JoinOn<Join<Alias<...>, ...,>,...>,...>`, due to diesel's fancy type system. And also I couldn't reuse the macros and aliases very nicely as it seems I have to define the aliases and macros inside each function for three times, if I want to use aliases. Therefore I don't think this refactoring is worth doing. Let me know if you think there's a nice way of doing it and I'll improve the code in a followup PR.